src/SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob/Updater/AcademicYearEndExpiryProcessor.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using SFA.DAS.Commitments.Domain.Data;
+using SFA.DAS.Commitments.Domain.Entities.DataLock;
+using SFA.DAS.Commitments.Domain.Interfaces;
+
+namespace SFA.DAS.Comitments.AcademicYearEndProcessor.WebJob.Updater
+{
+    public class AcademicYearEndExpiryProcessor : IAcademicYearEndExpiryProcessor
+    {
+        private readonly IAcademicYearDateProvider _academicYearProvider;
+        private readonly IDataLockRepository _dataLockRepository;
+        private readonly DataLockErrorCode _expirableErrorCodes;
+        private readonly ICurrentDateTime _currentDateTime;
+        private List<DataLockStatus> _expirableDatalocks = new List<DataLockStatus>();
+
+        public AcademicYearEndExpiryProcessor(IAcademicYearDateProvider academicYearProvider,
+            IDataLockRepository dataLockRepository, DataLockErrorCode expirableErrorCodes,
+            ICurrentDateTime currentDateTime)
+        {
+            _dataLockRepository = dataLockRepository ?? throw new ArgumentException("dataLockRepository");
```

Comment:
Couple of things. Firstly, throw only became an expression in C# 7, so this doesn't work in earlier versions. Perhaps we should all move to VS 2017 but I don't think we've "officially" said anything about that so probably needs changing back to an old fashioned if/then.

Secondly, we tend to use nameof(object) rather than a string when throwing ArgumentException.